src/FakePAB/PreBalance.hs

```diff
@@ -164,7 +174,7 @@ filterNonAda :: Value -> Value
 filterNonAda =
   mconcat
     . map unflattenValue
-    . filter (\(curSymbol, tokenName, _) -> curSymbol /= Ada.adaSymbol && tokenName /= Ada.adaToken)
+    . filter (\(curSymbol, tokenName, _) -> curSymbol /= Ada.adaSymbol || tokenName /= Ada.adaToken)
```

Comment:
[nits] I think the second half of this predicate is actually unnecessary, the currency symbol will decide if an asset is ada or not.

Reply:
Very true

Reply:
Simplified this and other ada checks, actually allowed me to simplify this function down to a simple key filter